example/src/fonts/Typo.tsx

```diff
@@ -0,0 +1,235 @@
+    letterSpacing: 1.5,
+  },
+  h5: {
+    fontFamily: 'Inter',
+    fontSize: 18,
+    lineHeight: 28,
+    letterSpacing: 1.5,
+  },
+});
+
+const HeadlinesSmall = StyleSheet.create({
+  h1: {
+    fontFamily: 'Inter',
+    fontSize: 42,
+    lineHeight: 48,
+    letterSpacing: 0.5,
+  },
+  h2: {
+    fontFamily: 'Inter',
+    fontSize: 36,
+    lineHeight: 42,
+    letterSpacing: 1,
+  },
+  h3: {
+    fontFamily: 'Inter',
+    fontSize: 24,
+    lineHeight: 32,
+    letterSpacing: 1,
+  },
+  h4: {
+    fontFamily: 'Inter',
+    fontSize: 20,
+    lineHeight: 32,
+    letterSpacing: 1.5,
+  },
+  h5: {
+    fontFamily: 'Inter',
+    fontSize: 18,
+    lineHeight: 28,
+    letterSpacing: 1.5,
+  },
+});
+
+const TextStyles = StyleSheet.create({
+  bodyBig: {
+    fontFamily: 'Inter',
+    fontSize: 20,
+    lineHeight: 36,
+  },
+  bodySmall: {
+    fontFamily: 'Inter',
+    fontSize: 16,
+    lineHeight: 28,
+  },
+  label: {
+    fontFamily: 'Inter',
+    fontSize: 12,
+    lineHeight: 16,
+  },
+  caption: {
+    fontFamily: 'Inter',
+    fontSize: 18,
+    fontWeight: 'bold',
+    lineHeight: 24,
+    letterSpacing: 1,
+    textTransform: 'uppercase',
+  },
+  button: {
+    fontFamily: 'Inter-Medium',
+    fontSize: 18,
+    fontWeight: 'bold',
+    lineHeight: 24,
+    letterSpacing: 1,
+  },
+});
+
+const TextStylesSmall = StyleSheet.create({
+  bodyBig: {
+    fontFamily: 'Inter',
+    fontSize: 18,
+    lineHeight: 32,
+  },
+  bodySmall: {
+    fontFamily: 'Inter',
+    fontSize: 16,
+    lineHeight: 28,
+  },
+  label: {
+    fontFamily: 'Inter',
+    fontSize: 12,
+    lineHeight: 16,
+  },
+  caption: {
+    fontFamily: 'Inter',
+    fontSize: 18,
+    fontWeight: 'bold',
+    lineHeight: 24,
+    letterSpacing: 3,
+    textTransform: 'uppercase',
+  },
+  button: {
+    fontFamily: 'Inter',
+    fontSize: 18,
+    fontWeight: 'bold',
+    lineHeight: 24,
+    letterSpacing: 3,
+  },
+});
+
+const TextStylesCustom = StyleSheet.create({
+  videoLabel: {
+    fontFamily: 'Inter',
+    fontSize: 14,
+    lineHeight: 18,
+  },
+  chatRegular: {
+    fontFamily: 'Inter',
+    fontSize: 14,
+    lineHeight: 21,
+  },
+  chatSemibold: {
+    fontFamily: 'Inter',
+    fontSize: 14,
+    lineHeight: 21,
+    fontWeight: 'bold',
+  },
+  chatTitle: {
+    fontFamily: 'Inter',
+    fontSize: 16,
+    lineHeight: 24,
+    letterSpacing: 3,
+    fontWeight: 'bold',
+  },
+});
+
+type VariantName =
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'body-big'
+  | 'body-small'
+  | 'label'
+  | 'caption'
+  | 'button'
+  | 'video-label'
+  | 'chat-regular'
+  | 'chat-semibold'
+  | 'chat-title';
+
+type TypoProps = {
+  variant: VariantName;
+  children: ReactNode;
+};
+
+export const Typo = (props: TypoProps) => {
+  const windowDimensions = Dimensions.get('window');
+
+  const [dimensions, setDimensions] = useState({
+    window: windowDimensions,
+  });
+
+  useEffect(() => {
+    const subscription = Dimensions.addEventListener(
```

Comment:
Good point, thanks!

Reply:
What about the second part? The hook still subscribes to events under the hood. Also the breakpoints are on height or width of the screen?

Reply:
Sorry, I misunderstood your intention. Should be good now.

Reply:
I just assumed that width is better way to determine when to shrink the font size since text is generally wider than taller. I sent a question about that to the designer, but she is OOO until later today.